### PR TITLE
Exi structures integration

### PR DESCRIPTION
**Pull request by WhiteTPoison**

### CODE REVIEW

File: Source/Core/Core/Brawlback/BrawlbackUtility.cpp [hunk 1]
```diff
@@ -5,95 +5,43 @@
...
-                ASSERT(framedata->frame == frame);
-                ret = framedata;
-            }
-        }
-        #else
-        for (int i = ((int)queue.size())-1; i >= 0; i--) {
-            if (queue[i]->_playerFrameData.frame == frame) {
-                ret = queue[i].get();
-                break;
+    PlayerFrameData* findInPlayerFrameDataQueue(const PlayerFrameDataQueue& queue, u32 frame) {
```
Comment (rapito) [inline Source/Core/Core/Brawlback/BrawlbackUtility.cpp]: How often is this method going to be called? Wouldnt we benefit from moving away from this queue and making it a map instead that we can access directly by frame number?

File: Source/Core/Core/Brawlback/BrawlbackUtility.cpp [hunk 2]
```diff
@@ -5,95 +5,43 @@
 
 namespace Brawlback
 {
-
+    void Reset(RollbackInfo& _rollbackInfo)
+    {
+      _rollbackInfo.isUsingPredictedInputs = false;
```
Comment (rapito) [inline Source/Core/Core/Brawlback/BrawlbackUtility.cpp]: @JaredWhiteOne I dont see where this method is being invoked? Also it looks like it should be kept as part of the RollbackInfo class and just do _rollbackInfo.Reset()
Else this Reset method will look weird if you have multiple objects of different classes to reset at some point

File: Source/Core/Core/Brawlback/BrawlbackUtility.cpp [hunk 3]
```diff
@@ -5,95 +5,43 @@
 
 namespace Brawlback
 {
-
+    void Reset(RollbackInfo& _rollbackInfo)
```
Comment (eliasreid) [inline Source/Core/Core/Brawlback/BrawlbackUtility.cpp]: Is this function being used anywhere? I can't find any usages at a quick glance. 

If so I'd suggest a couple style changes:
- rename function to resetRollbackInfo
- remove underscore from arg name

Comment (WhiteTPoison) [reply Source/Core/Core/Brawlback/BrawlbackUtility.cpp]: I addressed your style changes in a push below.

File: Source/Core/Core/Brawlback/BrawlbackUtility.cpp [hunk 4]
```diff
@@ -5,15 +5,72 @@
 
 namespace Brawlback
 {
+    void ResetRollbackInfo(RollbackInfo& rollbackInfo)
```
Comment (eliasreid) [inline Source/Core/Core/Brawlback/BrawlbackUtility.cpp:8]: Name is better but.. AFAICT this function is not used anywhere. I think it should be removed unless it is actually used

File: Source/Core/Core/Brawlback/BrawlbackUtility.cpp [hunk 5]
```diff
@@ -5,15 +5,72 @@
...
+      return fletcher32_checksum((short*)sums.data(), sizeof(int) * sums.size());
+    }
 
-    Match::PlayerFrameDataImpl* findInPlayerFrameDataQueue(const PlayerFrameDataQueue& queue, u32 frame) {
-        Match::PlayerFrameDataImpl* ret = nullptr;
-        #if 0
+    PlayerFrameData* findInPlayerFrameDataQueue(const PlayerFrameDataQueue& queue, u32 frame)
+    {
+      PlayerFrameData* ret = nullptr;
+#if 0
```
Comment (eliasreid) [inline Source/Core/Core/Brawlback/BrawlbackUtility.cpp:73]: Not a fan of code hanging around behind a #if 0 like this.

Is this needs to be kept, at least use `#ifdef SOME_NAME` here instead and document somewhere how / when to use these switches

Comment (WhiteTPoison) [reply Source/Core/Core/Brawlback/BrawlbackUtility.cpp:73]: This is OOS for this PR. @FaultyPine Any way you can go in and fix after?

File: Source/Core/Core/Brawlback/BrawlbackUtility.h [hunk 6]
```diff
@@ -13,7 +13,7 @@
 #include "SlippiUtility.h"
 #include "Brawltypes.h"
 #include "Savestate.h"
-
+#include <Core/Brawlback/include/brawlback-exi-structures/ExiStructures.h>
```
Comment (eliasreid) [inline Source/Core/Core/Brawlback/BrawlbackUtility.h]: We should just have Core/Brawlback/include in the include path, then this line can become:
`#include <brawlback-exi-structures/ExiStructures.h>`

File: Source/Core/Core/Brawlback/BrawlbackUtility.h [hunk 7]
```diff
@@ -327,78 +327,12 @@ namespace Brawlback {
...
-        struct PlayerSettingsImpl
-        {
-            PlayerSettings _playerSettings;
-        };
-
-        struct GameSettingsImpl
-        {
-            GameSettings _gameSettings;
-        };
         struct Game {
```
Comment (eliasreid) [inline Source/Core/Core/Brawlback/BrawlbackUtility.h]: Maybe my visual studio search skills are terrible.. but is this struct Game actually used anywhere? When I comment it out, the project still builds fine

File: Source/Core/DolphinLib.vcxproj [hunk 8]
```diff
@@ -52,8 +52,17 @@
     <ClCompile Include="Core\HW\EXI\EXIBrawlback.cpp" />
   </ItemGroup>
   <ItemGroup>
+    <ClInclude Include="Core\Brawlback\include\brawlback-exi-structures\BrawlbackPad.h" />
     <ClInclude Include="Core\Brawlback\include\brawlback-exi-structures\exi-structures.h" />
```
Comment (eliasreid) [inline Source/Core/DolphinLib.vcxproj:57]: This include file no longer exists in exi-structures repo

**Review by eliasreid — APPROVED** [review]

**Pull request merged**